### PR TITLE
Set pilosa endpoint through an environment variable.

### LINKED ISSUE #315: boltdb file apparently is not created for indexes when gitbase run inside a container
Running gitbase from the image present in docker hub, I can't see the boltdb file inside the container when an index is created.

### PR DESCRIPTION
**Pull request by mcarmonaa**
Closes #315

Gitbase wasn't connecting to the pilosa container properly.

I made the pilosa endpoint configurable through and environment variable too to make easier run gitbase with docker.

Now it's documented too in the README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -99,16 +100,34 @@ Available commands:
...
+```
+docker run --rm --name gitbase -p 3306:3306 --link pilosa:pilosa -e PILOSA_ENDPOINT="http://pilosa:10101" -v /my/git/repos:/opt/repos srcd/gitbase:latest
+```
+
+### Client
 A MySQL client is needed to connect to the server. For example:
 
 ```bash
-$ mysql -q -u root -h 127.0.0.1
+$ mysql -q -u root -h 127.0.0.1 -u gitbase
```
Comment (jfontan) [inline README.md]: Why changing the user's name? Besides now `-u` is set twice.

Comment (mcarmonaa) [reply README.md]: The default user is changed in the Dockerfile to `gitbase`. I'm changing this to reflect what user you should use depending on gitbase running in local or docker.

**Comment by jfontan** [comment]
The default user should be the same in docker or standalone mode at some point.

**Comment by ajnavarro** [comment]
@mcarmonaa can you rebase please?

**Comment by mcarmonaa** [comment]
@ajnavarro done

**Pull request merged**